API/pubsub/README.md

```diff
@@ -0,0 +1,63 @@
+pubsub API
+==========
+
+#### `pubsub.subscribe`
+
+> Subscribe to an IPFS Topic
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.pubsub.subscribe(topic, options, callback)
+
+- `topic` - type: String
+- `options` - type: Object, optional, might contain the following properties:
+  - `discover`: type: Boolean - Will use the DHT to find 
+
+`callback` must follow `function (err, subscription) {}` where Subscription is a Node.js Stream in Object mode, emiting a `data` event for each new message on the subscribed topic.`err` is an error if the operation was not successful.
```

Comment:
I don't think this is a good interface. We all know the issues of node.js streams and using object streams is even more problematic, as some modules (like hapi.js) don't even work with object streams.

I would propose a simpler api using raw `EventEmitter`.

```js
const handler = onMessage
ipfs.pubsub.subscribe('mytopic', {discover: true}, onMessage)
ipfs.pubsub.unsubscribe('mytopic', onMessage)
```

Which would be implemented using `EventEmitter` and mostly aliases for `.on` and `removeListener`. Where `removeListener` does a check onto [emitter.listenerCount(topic)](https://nodejs.org/docs/latest/api/events.html#events_emitter_listenercount_eventname) and if that is `0` it will actually unsubscribe from the underlying thing.